network/datadog_checks/network/network.py

```diff
@@ -413,29 +416,46 @@ def _check_linux(self, instance):
                         nstat_metrics_names[k][met], self._parse_value(netstat_data[k][met]), tags=custom_tags
                     )
 
-        proc_conntrack_path = "{}/net/nf_conntrack".format(proc_location)
-        try:
-            with open(proc_conntrack_path, 'r') as conntrack_file:
-                # Starting at 0 as the last line has a line return
-                conntrack_count = 0
-                while 1:
-                    # Reading the file by chucks (64k being a randomly chosen buffer size)
-                    conntrack_buffer = conntrack_file.read(65536)
-                    if not conntrack_buffer:
-                        break
-                    conntrack_count += conntrack_buffer.count('\n')
-                self.gauge('system.net.conntrack.count', conntrack_count, tags=custom_tags)
-        except IOError:
-            self.log.debug("Unable to read %s. Skipping conntrack metrics pull.", proc_conntrack_path)
-
-        proc_conntrack_max_path = "{}/sys/net/nf_conntrack_max".format(proc_location)
-        try:
-            with open(proc_conntrack_max_path, 'r') as conntrack_max_file:
-                # Starting at 0 as the last line has a line return
-                conntrack_max = conntrack_max_file.read().rstrip()
-                self.gauge('system.net.conntrack.max', conntrack_max, tags=custom_tags)
-        except IOError:
-            self.log.debug("Unable to read %s. Skipping nf_conntrack_max metrics pull.", proc_conntrack_max_path)
+        conntrack = pyroute2.conntrack.Conntrack()
+        for cpu in conntrack.stat():
+            """
+            [
+                {
+                    'cpu': 0, 'found': 27644, 'invalid': 19060, 'ignore': 485633411, 'insert': 0,
+                    'insert_failed': 1, 'drop': 1, 'early_drop': 0, 'error': 0, 'search_restart': 39936711
+                },
+                {
+                    'cpu': 1, 'found': 21960, 'invalid': 17288, 'ignore': 475938848, 'insert': 0,
+                    'insert_failed': 1, 'drop': 1, 'early_drop': 0, 'error': 0, 'search_restart': 36983181
+                },
+            ]
+            """
+            cpu_tag = ['cpu:{}'.format(cpu['cpu'])]
+            for metric, value in iteritems(cpu):
+                if metric != 'cpu':
+                    self.monotonic_count('system.net.conntrack.{}'.format(metric), value, tags=custom_tags + cpu_tag)
```

Comment:
I'm not sure tagging with cpu # makes sense
But it doesn't hurt to have it and we have all the data this way